### DISCUSSION TITLE
lvgl_porting what does it do

### DISCUSSION THREAD

**Discussion by sjgcornish** [Westcott1/Waveshare-ESP32-S3-Touch-LCD-4.3-and-Arduino#2] [General]
Hi
I have eventually managed to get the widgets demo working with the LCD-4.3 using Arduino and the wave share lvgl_porting but I am a bit confused as to the role of the lvgl_porting and do I always need to use it for any of the programs I write and the difference and or relationship  between TFT_espi and lvgl. Does TFT_espi.h use LVGL use or use LVGLv use TFT_espi.h.
The TFT_espi.h which does not support the ST7262 and I the demos do not #inlcude tft_espi.h so again I assume that lvgl does not need tft_espi.h to work and the lvgl_porting enables programs written using LVGL (without tft_espi.h) to work on the ST7262 i.e. on this device.  Does this mean then that your Arduino example has to be plugged in to the waveshare lvgl_porting.

**Comment by Westcott1** [comment]
Hi Sjgcornish,
TFT_espi is needed to handle all of the basic graphics. That's why I had to develop the pin mapping for this particular board.
The Arduino_GFX example shows this.
LVGL, when used, sits on top, needing TFT_espi  to display its fancy graphics using the DisplayFlush function.
It cannot function on its own.
My Arduino_LVGL example is a complete sketch, nothing more is needed.

These examples use LVGL v8.x and Arduino esp32 v2.x.
Neither LVGL v9.x nor Arduino esp32 v3.x will work (yet).

**Reply by sjgcornish** [reply]
Many thanks for your quick reply. On further research, as far as I can see,  you can use lvgl with out tft-espi by using ESP_DisplayPanel and lvg_porting. The examples provided by waveshare do not include tft_espi.h (unless lvgl itself includes it)

**Reply by Westcott1** [reply]
Yes, ESP_DisplayPanel  is Waveshare's customised/hacked version of TFT_eSPI !

**Reply by sjgcornish** [reply]
All is now clear, Thank you. I will be interested in seeing how your development move forwards

**Comment by Westcott1** [comment]
Hi Sjgcornish,
I'm using the UART port for uploading and Serial.
Both work OK
ESP-ROM:esp32s3-20210327 is the first line printed after booting.
I've seen this behaviour before, but can't remember why.
It's not just this board.

**Comment by sjgcornish** [comment]
Hi Hamish

I found that I can only upload on the the USB port which displays
ESP-ROM:esp32s3-20210327.  The UART port supports both upload and
serial.print as expected

Regards
Steve Cornish

**Comment by sjgcornish** [comment]
PS Do you happen to know what the Battery voltage should be 3.7v does not
appear to work so I guess it would be a 2S or 7.4V but I do not have one to
try.

Regards

Steve

**Reply by Westcott1** [reply]
According to the Waveshare connection diagram, connector 18 is "3.7V single lithium battery PH2.0 header"

**Comment by sjgcornish** [comment]
Thank you

Steve

**Comment by sjgcornish** [comment]
Hi Again

Sony for troubling you again. I have found that my UI does not always come
up when connecting power either via the battery either of the usb ports. If
I bring up the panel with usb power then turn on battery it is maintained
when disconnecting the usb port. But turning off battery then back on again
the ui does not come up. My program is running when the ui is not showing
as I get an output from the serial port. The backlight is on but no ui is
displayed. I am not sure if this is a hardware problem or software. Using
Lvgl v8.4. Any ideas have you seen this. Waveshaee support not much help.

Regards

Steve

**Reply by Westcott1** [reply]
Hi Steve,
Sorry, can't help here.
I only ever use USB power.
